src/Adapter/Carrier/Repository/CarrierRangeRepository.php

```diff
@@ -0,0 +1,213 @@
+
+        // Then, create query to retrieve carrier ranges
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('cd.id_zone, cr.delimiter1 AS range_from, cr.delimiter2 AS range_to, cd.price AS range_price')
+            ->from($this->dbPrefix . 'delivery', 'cd')
+            ->andWhere('cd.id_carrier = :carrierId')
+            ->setParameter('carrierId', $carrierId->getValue());
+
+        // Apply shipping method (weight or price) and shop constraint, then execute the query
+        $this->applyRangeTypeForQuery($qb, $carrier);
+        $this->applyShopConstraint($qb, $shopConstraint);
+        $results = $qb->executeQuery()->fetchAllAssociative();
+
+        return $results;
+    }
+
+    /**
+     * @param CarrierId $carrierId
+     * @param CarrierRangesCollectionInterface $rangesCollection
+     * @param ShopConstraint $shopConstraint
+     */
+    public function set(CarrierId $carrierId, CarrierRangesCollectionInterface $rangesCollection, ShopConstraint $shopConstraint): void
+    {
+        // Check shop constraint
+        $this->assertShopConstraint($shopConstraint);
+
+        // Get carrier
+        $rangeTable = $this->getRangeMethodTable($rangesCollection->getCalculatingMethod());
+
+        // Use transaction to ensure data consistency
+        $this->connection->beginTransaction();
+
+        // Reset carrier ranges
+        $this->reset($carrierId, $shopConstraint);
+
+        // Foreach zones
+        $rangesIds = [];
+        foreach ($rangesCollection->getZones() as $zone) {
+            $idZone = $zone->getZoneId();
+
+            // Foreach ranges in current zone
+            foreach ($zone->getRanges() as $range) {
+                // Check if range already exist in database
+                if (!in_array($range->getFrom() . '-' . $range->getTo(), array_keys($rangesIds), true)) {
+                    // Insert range
+                    $this->connection->insert(
+                        $this->dbPrefix . $rangeTable,
+                        [
+                            'id_carrier' => $carrierId->getValue(),
+                            'delimiter1' => $range->getFrom(),
+                            'delimiter2' => $range->getTo(),
+                        ]
+                    );
+                    $rangeId = $this->connection->lastInsertId();
+                    $rangesIds[$range->getFrom() . '-' . $range->getTo()] = $rangeId;
+                } else {
+                    $rangeId = $rangesIds[$range->getFrom() . '-' . $range->getTo()];
+                }
+
+                // Insert price in delivery table
+                $this->connection->insert(
+                    $this->dbPrefix . 'delivery',
+                    [
+                        'id_carrier' => $carrierId->getValue(),
+                        'id_' . $rangeTable => $rangeId,
+                        'id_zone' => $idZone,
+                        'price' => $range->getPrice(),
```

Comment:
I think we should explicitly set this even if we only handle All shops for now (and even if it's the default value), so that later we already understand where we'll need to vary this data for full multishop support
```suggestion
                        'price' => $range->getPrice(),
                        // Only all shops is handled for now
                        'id_shop' => null,
                        'id_shop_group' => null,
```